agent/connect/ca/provider_consul_test.go

```diff
@@ -35,7 +35,7 @@ func (c *consulCAMockDelegate) ApplyCARequest(req *structs.CARequest) (interface
 
 		return true, nil
 	case structs.CAOpDeleteProviderState:
-		if err := c.state.CADeleteProviderState(req.ProviderState.ID); err != nil {
+		if err := c.state.CADeleteProviderState(idx+1, req.ProviderState.ID); err != nil {
```

Comment:
Unrelated drive-by fix?

Reply:
No - we now require a raft index for every FSM operation the mutates state since `WriteTxn` requires one to capture it for event stuff.

So there were a few places where our state store methods performed writes but didn't take an index param because they didn't use it before and I had to change that.